yt_dlp/extractor/nzonscreen.py

```diff
@@ -0,0 +1,68 @@
+from yt_dlp.extractor.common import InfoExtractor
+from yt_dlp.utils import (
+    ExtractorError,
+    traverse_obj,
+)
+
+
+class NZOnScreenIE(InfoExtractor):
+    _VALID_URL = r'^https://www.nzonscreen.com/title/(?P<id>[^\?]+)'
+    _TESTS = [{
+        'url': 'https://www.nzonscreen.com/title/shoop-shoop-diddy-wop-cumma-cumma-wang-dang-1982',
+        'md5': 'a0375e9b6f14b235438506cb0aee974c',
+        'info_dict': {
+            'id': 'shoop-shoop-diddy-wop-cumma-cumma-wang-dang-1982',
+            'ext': 'mp4',
+            'title': 'Monte Video - "Shoop Shoop, Diddy Wop"',
+            'format_id': 'lo_res'
+        },
+    }, {
+        'url': 'https://www.nzonscreen.com/title/shes-a-mod-1964?collection=best-of-the-60s',
+        'md5': 'e8a56d7d02cdcdc5d6f8897e2532e157',
+        'info_dict': {
+            'id': 'shes-a-mod-1964',
+            'ext': 'mp4',
+            'title': 'Ray Columbus - \'She\'s A Mod\'',
+            'format_id': 'lo_res'
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        playlist = self._parse_json(self._html_search_regex(
+            r'data-video-config=\'(.+?)\'', webpage, 'media data'),
+            video_id)
+
+        title = traverse_obj(playlist, 'label') or traverse_obj(playlist, 'description') or ''
+        if not title:
+            page_title = self._html_extract_title(webpage, default=None) or self._og_search_title(
+                webpage, default=None)
+            if page_title and '|' in page_title:
```

Comment:
I was being overly cautious with the `|` detection -- it is always present, as far as I can see, on most pages. But I wanted to make sure that it was present i.e. that the title format was the one that I was expecting (`{video title} | {collection} | NZ On Screen`), before calling `.split` and taking the first element.

If it's enough to only take the title from the m3u8, then maybe that's an easier option. I just thought this may be a useful fallback.

Reply:
playlist.label, playlist.description and title from webpage are all giving different values. So I've put them in separate fields

Reply:
I just saw that, thank you!